chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -242,7 +243,14 @@ public Octree(int octreeDepth, Node node) {
    * @param type The new voxel type to be set
    */
   public synchronized void set(int type, int x, int y, int z) {
-    set(new Node(type), x, y, z);
+    try {
+      implementation.set(type, x, y, z);
+    } catch(PackedOctree.OctreeTooBigException e) {
+      // Octree is too big, switch implementation and retry
+      Log.warn("Octree is too big, falling back to old (slower and bigger) implementation.");
+      switchToNodeBased();
```

Comment:
Oh... this wouldn't be possible with the changes I suggested above (regarding the interfaces).